--- libupnp/configure.ac.orig	2011-02-10 23:53:25.000000000 +0100
+++ libupnp/configure.ac	2011-02-10 23:54:23.574454501 +0100
@@ -546,6 +546,7 @@
 AC_SUBST(PTHREAD_CFLAGS)
 AC_SUBST(PTHREAD_CC)
 
+AC_DEFINE([_WIN32_WINNT], 0x0501, [Define to '0x0500' for Windows 2000 APIs.])
 #
 # Determine if pthread_rwlock_t is available
 #
--- libupnp/libupnp.pc.in	2010-12-23 21:24:05.000000000 +0100
+++ libupnp.new/libupnp.pc.in	2011-02-13 11:27:23.000000000 +0100
@@ -6,6 +6,6 @@
 Name: libupnp
 Description: Linux SDK for UPnP Devices
 Version: @VERSION@
-Libs: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ -L${libdir} -lupnp -lthreadutil -lixml 
+Libs: @PTHREAD_CFLAGS@ -L${libdir} -lupnp -lthreadutil -lixml -liphlpapi @PTHREAD_LIBS@ 
 Cflags: @PTHREAD_CFLAGS@ -I${includedir}/upnp
 
--- libupnp/upnp/src/inc/upnputil.h	2010-12-23 21:24:06.000000000 +0100
+++ libupnp.new/upnp/src/inc/upnputil.h	2011-02-13 08:24:24.000000000 +0100
@@ -125,7 +125,7 @@
 		#define strncasecmp		strnicmp
 		#define sleep(a)		Sleep((a)*1000)
 		#define usleep(a)		Sleep((a)/1000)
-		#define strerror_r(a,b,c)	(strerror_s((b),(c),(a)))
+		#define strerror_r(a,b,c)       strncpy( b, strerror(a), c)
 	#else
 		#define max(a, b)   (((a)>(b))? (a):(b))
 		#define min(a, b)   (((a)<(b))? (a):(b))
--- upnp/upnp/inc/UpnpInet.h	2011-04-03 04:50:36.000000000 +0200
+++ upnp.neww/upnp/inc/UpnpInet.h	2011-11-18 01:54:45.418529337 +0100
@@ -15,11 +15,6 @@
 
 #ifdef WIN32
 	#include <stdarg.h>
-	#ifndef UPNP_USE_MSVCPP
-		/* Removed: not required (and cause compilation issues) */
-		#include <winbase.h>
-		#include <windef.h>
-	#endif
 	#include <winsock2.h>
 	#include <iphlpapi.h>
 	#include <ws2tcpip.h>
--- libupnp/threadutil/inc/ThreadPool.h	2011-01-20 07:46:57.000000000 +0100
+++ libupnp.new/threadutil/inc/ThreadPool.h	2011-09-23 01:36:12.000000000 +0200
@@ -45,6 +45,7 @@
 #include <errno.h>

 #ifdef WIN32
+	#ifndef _TIMEZONE_DEFINED
	#include <time.h>
	struct timezone
	{
@@ -52,6 +53,7 @@
		int  tz_dsttime;     /* type of dst correction */
	};
	int gettimeofday(struct timeval *tv, struct timezone *tz);
+	#endif
 #else /* WIN32 */
	#include <sys/param.h>
	#include <sys/time.h> /* for gettimeofday() */
--- libupnp/upnp/inc/upnp.h.orig	2012-03-22 00:15:38.000000000 +0100
+++ libupnp/upnp/inc/upnp.h	2012-03-28 18:58:55.043642000 +0200
@@ -61,6 +61,20 @@
	/* Other systems ??? */
 #endif

+#   if defined( __MINGW32__ )
+#       if !defined( _OFF_T_ )
+            typedef long long _off_t;
+            typedef _off_t off_t;
+#           define _OFF_T_
+#       else
+#           ifdef off_t
+#               undef off_t
+#           endif
+#           define off_t long long
+#       endif
+#   endif
+
+
 #define LINE_SIZE  (size_t)180
 #define NAME_SIZE  (size_t)256
 #define MNFT_NAME_SIZE  64
