--- harfbuzz-0.9.10/configure.ac.orig	2013-01-03 15:08:51.000000000 +0900
+++ harfbuzz-0.9.10/configure.ac	2013-01-10 22:06:26.000000000 +0900
@@ -117,7 +117,7 @@
 
 dnl ===========================================================================
 
-have_hb_old=true
+have_hb_old=false
 if $have_hb_old; then
 	AC_DEFINE(HAVE_HB_OLD, 1, [Have Old HarfBuzz backend])
 fi
@@ -221,6 +221,7 @@
	[have_uniscribe=true; AC_MSG_RESULT(yes)],
	[have_uniscribe=false;AC_MSG_RESULT(no)])
 LIBS=$saved_LIBS
+have_uniscribe=false
 if $have_uniscribe; then
 	UNISCRIBE_CFLAGS=
 	UNISCRIBE_LIBS="-lusp10 -lgdi32"
@@ -233,6 +234,7 @@
 dnl ===========================================================================
 
 AC_CHECK_TYPE(CTFontRef, have_coretext=true, have_coretext=false, [#include <ApplicationServices/ApplicationServices.h>])
+have_coretext=false
 if $have_coretext; then
 	CORETEXT_CFLAGS=
 	CORETEXT_LIBS="-framework ApplicationServices"
